examples/worklet.rs

```diff
@@ -143,13 +143,17 @@ impl AudioProcessor for WhiteNoiseProcessor {
 
     fn onmessage(&mut self, msg: Box<dyn std::any::Any + Send + 'static>) {
         // handle incoming signals requesting for change of color
-        if let Some(&color) = msg.downcast_ref::<NoiseColor>() {
-            self.color = color;
-            return;
-        }
+        let msg = match msg.downcast::<NoiseColor>() {
+            Ok(color) => {
+                self.color = *color;
+                return;
+            }
+            Err(msg) => msg,
+        };
 
-        // ignore other incoming messages
-        log::warn!("WhiteNoiseProcessor: Ignoring incoming message");
+        // ...add more message handlers here...
+
+        log::warn!("WhiteNoiseProcessor: Dropping incoming message {msg:?}");
```

Comment:
Would be convenient for debugging, but I haven't tried it yet to see what the consequences are.

Those scattered and redundant debug logs will be replaced by a single log in the next refactoring step. Let's not worry about this now.

Reply:
This turns out to be more difficult than expected. Looks like we need to change the `onmessage()` API once again, because destructive consumption by using `downcast` does not work then. See also: [as-any](https://docs.rs/as-any/latest/as_any/) or [downcast-rs](https://docs.rs/downcast-rs/latest/downcast_rs/).

Reply:
What exactly would you need these crates for? How are you trying to unify the debug logs?

I was thinking, perhaps we can prevent the deallocation in `onmessage` by preventing users to move out of the Box, with this signature:

```rust
    fn onmessage(&mut self, msg: &mut dyn std::any::Any>)
```

And then we can ship the handled messages to a deallocation thread.
This would be quite trivial to implement.

Reply:
Yes, that's basically the idea. The message handlers must be able to replace the content of the incoming message with stuff that needs to be deallocated outside of the render thread.

Reply:
If you agree with this approach I could try that. In the first step the deallocation will still happen in `route_message()` in the render thread. This could be resolved later.

The message handlers are responsible for preventing deallocations themselves, i.e. using `downcast_mut()` and `std::mem::replace()` as needed.

Reply:
You can't box `dyn std::any::Any + std::fmt::Debug`. Instead you need to introduce a new helper trait that requires `std::any::Any + std::fmt::Debug`. But then you can't downcast the boxed type directly. That's when `AsAny` comes into play.

Reply:
Yes, agreed on all points. Looking forward to your initial implementation